source/code/scxsystemlib/common/scxsysteminfo.cpp

```diff
@@ -675,6 +679,16 @@ namespace SCXSystemLib
     }
 
     /**
+           Returns the default tmp directory path for this platform.
+           RunAs providers Executescript will store the scripts at this location.
+
+           \returns Default path for the temp dir
+    */
+    string SystemInfo::GetDefaultTmpDir() const
+    {
+    	return s_defaultTmpDir;
```

Comment:
I don't see a need to return this. Just implement "GetTemporaryDirectory()" that takes the default into account (default check whatever was passed in "SetTemporaryDirectory()" and, if that wasn't called, then default to wstring `/tmp/`.